### ISSUE TITLE
Problem with SystemChrome.setPreferredOrientations when I use it with an iPad with iOS 16

### ISSUE THREAD

**Issue by csacchetti** [flutter/flutter#121007]
I am forced to reopen #119473  because it is indeed still there. 

In my project, I only need to have one page in landscape mode and not in portrait mode, while the rest of the pages are in portrait and landscape mode.

As per the documentation, I used SystemChrome.setPreferredOrientations
```
@override 
   void initState() {
    
    // set the screen in landscape mode
    SystemChrome.setPreferredOrientations([
      DeviceOrientation.landscapeLeft,
    ]);

   super.initState();
}
and
```
```
@override
  void dispose() {
    // restore the
    SystemChrome.setPreferredOrientations([
      DeviceOrientation.portraitUp,
      DeviceOrientation.landscapeRight,
      DeviceOrientation.landscapeLeft,
    ]);
    
    super.dispose();
  }
```
when I exit the page to restore all the orientations.

If you remove "Stage manager" from the iPad and the iPad works as before on full screen. After introducing "required full screen" function in xCode in runner/general you can see that the error no longer does this, but if you force a single page to remain in landscape with
```
 SystemChrome.setPreferredOrientations([
      DeviceOrientation.landscapeRight,
    ]);
```
when you return to the rest of the pages it does not take the various orientations,
```
SystemChrome.setPreferredOrientations([
      DeviceOrientation.portraitUp,
      DeviceOrientation.landscapeRight,
      DeviceOrientation.landscapeLeft,
    ]);
```
but only takes one of them (e.g. only portraitUp or only landscapeLeft etc.).

Before entering the page and running the command that forces landscape mode all orientations were available as set in the info.plist. I think this is a bug that arose with the change in page management in ipadOS 16 due to the introduction of 'Stage manager'. It needs to be corrected as soon as possible because this is a basic function.

My Flutter doctor -v
```
Flutter (Channel stable, 3.7.1, on macOS 13.2 22D49 darwin-arm64, locale it-IT)
    • Flutter version 3.7.1 on channel stable at /Users/carlosacchetti/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (2 weeks ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/carlosacchetti/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.75.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (4 available)
    • iPhone di Carlo (mobile) • 8197a7889687e1b6883fab3f8b749fd17b32538b • ios            • iOS 16.2 20C65
    • iPad di CS (mobile)      • 00008027-001C504E1168002E                • ios            • iOS 16.3.1 20D67
    • macOS (desktop)          • macos                                    • darwin-arm64   • macOS 13.2 22D49 darwin-arm64
    • Chrome (web)             • chrome                                   • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by andychucs** [comment]
I have nearly issue on iOS with Flutter 3.7.3
Looks like flutter have not fixed on stable channel.

    • Flutter version 3.7.3 on channel stable at /Users/andy/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9944297138 (10 days ago), 2023-02-08 15:46:04 -0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1

**Comment by csacchetti** [comment]
Since this is an essential function, if indeed the problem has been solved in the 'master', I would recommend merging with the 'stable' version, otherwise the publication of apps will be blocked. Other bugs can be left but this one I would say no.

**Comment by exaby73** [comment]
Hello @csacchetti. Is this issue reproducible on the master channel?

**Comment by csacchetti** [comment]
I am in a state of production at the moment and prefer to stay on the 'stable' version, so I have not been able to test whether it has been resolved in the master. I had previously been told that the problem had been addressed and solved in the master, so I was waiting for at least this fix to be merged into the 'stable' channel.

**Comment by exaby73** [comment]
Based on https://github.com/flutter/flutter/issues/119473, looks like the fix is in master. You can request a cherry-pick into the stable channel. You can find more information [here](https://github.com/flutter/flutter/wiki/Flutter-Cherrypick-Process). But since this issue is fixed, I can closing this issue. Thank you

**Comment by csacchetti** [comment]
Can you give the "Commit Hash" and the "pr_link" of the master so I can put them in the Cherry-pick request. Thanks

**Comment by csacchetti** [comment]
I think I found them:
6ea01e6
https://github.com/flutter/flutter/blob/stable/bin/internal/release-candidate-branch.version

**Comment by Eimji** [comment]
Hello,

I have a similar issue on iPhone and iPad, the issue is not fixed in the latest stable channel 3.7.4.

Please a cherry-pick is really needed for the next stable version.

Thanks
